compiler/qsc_rir/src/passes/reindex_qubits.rs

```diff
@@ -47,7 +47,7 @@ pub fn reindex_qubits(program: &mut Program) {
         used_cx,
         cx_id,
         mresetz_id,
-        highest_used_id: program.num_qubits - 1,
+        highest_used_id: program.num_qubits.max(1).sub(1),
```

Comment:
nit: adding a small comment here about this being needed for cases in which no qubits are used would be useful because it isn't evident just by looking at the use of `max`.

Reply:
Added comment here to clarify, thanks!